src/hv_exc.c

```diff
@@ -111,10 +113,25 @@ static void hv_maybe_switch_cpu(struct exc_info *ctx, uartproxy_boot_reason_t re
 void hv_exc_proxy(struct exc_info *ctx, uartproxy_boot_reason_t reason, u32 type, void *extra)
 {
     /*
-     * If we end up in the proxy due to an event while the host is trying to switch CPUs,
-     * handle it as a CPU switch first. We still tell the host the real reason code, though.
+     * Wait until other CPUs finish processing requests or this CPU gets requested.
+     * If a CPU switch in rendezvous is requested, handle it before actually handling the
+     * exception. We still tell the host the real reason code, though.
      */
-    hv_maybe_switch_cpu(ctx, reason, type, extra);
```

Comment:
We can keep this here, because it won't do anything if `hv_want_cpu` is false...